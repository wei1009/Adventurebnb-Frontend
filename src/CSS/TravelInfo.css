.home-Info-detail{
    text-align: center;
}

.city-destination-recommand{
    text-align: center;
    margin:0.5rem 0 2rem 0;
}

.destination-title{
    margin-bottom:1.5rem;
}

.city-layout{
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.city-destination{
    width:147px;
    padding:2px;
    margin:3px;
    text-decoration: none;
}

.city-destination-image{
    border-radius:50%;
    width:124px;
    height:124px;
    padding:0;
    background-repeat:no-repeat ;
    background-position:center ;
    background-size: cover;
    margin:0 auto;
}

.home-grid {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr; 
    grid-gap:2rem;
    min-height:50rem;
}

.home-grid-item-1{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    background-color: rgb(250, 235, 215, 0.5);
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 1;
    grid-row-end: 2;
    display: flex;
    align-items: center;
    justify-content: center;
}



.home-grid-item-2{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    grid-column-start: 2; 
    grid-column-end: 3;
    grid-row-start: 1;
    grid-row-end: 3;
    background-repeat:no-repeat ;
    background-position:center ;
    background-size: cover;
}

.home-grid-item-3{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    background-color: rgb(250, 235, 215, 0.5);
    grid-column-start: 3;
    grid-column-end: 4;
    grid-row-start: 1;
    grid-row-end: 2;
    display: flex;
    align-items: center;
    justify-content: center;
}

.home-grid-item-4{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    grid-column-start:1;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 4;
    background-repeat:no-repeat ;
    background-position:center ;
    background-size: cover;
}

.home-grid-item-5{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    background-color: rgb(250, 235, 215, 0.5);
    grid-column-start: 2;
    grid-column-end: 3;
    grid-row-start: 3;
    grid-row-end: 4;
    display: flex;
    align-items: center;
    justify-content: center;
}

.home-grid-item-6{
    /* border: 1px solid rgba(126, 126, 125, 0.8); */
    grid-column-start: 3;
    grid-column-end: 4;
    grid-row-start: 2;
    grid-row-end: 4;
    background-repeat:no-repeat ;
    background-position:center ;
    background-size: cover;
}

@media(max-width: 1200px) and (min-width: 768px){
    .home-grid {
        display: grid;
        grid-template-columns: 1fr 1fr ; 
        grid-gap:1rem;
        /* min-height:45rem; */
    }
    
    .home-grid-item-1{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column-start: 1;
        grid-column-end: 2;
        grid-row:1;
        display: flex;
        align-items: center;
        justify-content: center;
        min-height:25rem;
    }
    
    .home-grid-item-2{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column-start: 2; 
        grid-column-end: 3;
        grid-row:1;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
    
    .home-grid-item-3{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column-start:2;
        grid-column-end: 3;
        grid-row:2;
        min-height:25rem;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
    
    .home-grid-item-4{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column-start: 1;
        grid-column-end: 2;
        grid-row:2;
        display: flex;
        align-items: center;
        justify-content: center;
       
    }
    
    .home-grid-item-5{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column-start: 1;
        grid-column-end: 2;
        grid-row:3;
        display: flex;
        align-items: center;
        justify-content: center;
        min-height:25rem;
    }
    
    .home-grid-item-6{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column-start: 2;
        grid-column-end: 3;
        grid-row:3;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
}

@media(max-width: 768px){
    .home-grid {
        display: grid;
        grid-template-columns: 100% ;
        grid-gap:0.5rem;
    }
    
    .home-grid-item-1{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column:1;
        grid-row:1;
        display: flex;
        align-items: center;
        justify-content: center;
    }
    
    .home-grid-item-2{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column:1;
        grid-row:2;
        min-height:25rem;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
    
    .home-grid-item-3{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column:1;
        grid-row:3;
        display: flex;
        align-items: center;
        justify-content: center;
    }
    
    .home-grid-item-4{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column:1;
        grid-row:4;min-height:25rem;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
    
    .home-grid-item-5{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        background-color: rgb(250, 235, 215, 0.5);
        grid-column:1;
        grid-row:5;
        display: flex;
        align-items: center;
        justify-content: center;
    }
    
    .home-grid-item-6{
        /* border: 1px solid rgba(126, 126, 125, 0.8); */
        grid-column:1;
        grid-row:6;
        min-height:25rem;
        background-repeat:no-repeat ;
        background-position:center ;
        background-size: cover;
    }
}