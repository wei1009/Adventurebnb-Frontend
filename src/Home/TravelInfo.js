import React from "react";
import { Link } from "react-router-dom";
import "../CSS/TravelInfo.css"

function TravelInfo() {
    const date = new Date();
    let checkInDate = date.toISOString().split('T')[0]
    let defaultCheckOutDate = new Date(checkInDate);
    defaultCheckOutDate.setDate(defaultCheckOutDate.getDate() + 1)
    let checkOutDate = defaultCheckOutDate.toISOString().split('T')[0];

    return (
        <div className="home-Info-detail">
            <div className="city-destination-recommand">
                <div className="container">
                    <h2 className="destination-title">Top destinations in the United States</h2>
                    <div className="city-layout column">

                        <Link className="city-destination row " 
                        to={`/city?type=city&city_code=NEW+YORK&state_code=NY&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1499092346589-b9b6be3e94b2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80)` }}>

                            </img>
                            <span>New York (NY)</span>
                        </Link>
                        <Link className="city-destination row "
                        to={`city?type=city&city_code=LAS+VEGAS&state_code=NV&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1470076892663-af684e5e15af?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1016&q=80)` }}>
                            </img>
                            <span>Las Vegas (NV)</span>
                        </Link>
                        <Link className="city-destination row "
                        to={`city?type=city&city_code=LOS+ANGELES&state_code=CA&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1572975165675-ab322e4b55ba?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80)` }}>
                            </img>
                            <span>Los Angeles (CA)</span>
                        </Link>
                        <Link className="city-destination row "
                        to={`city?type=city&city_code=ORLANDO&state_code=FL&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1575089776834-8be34696ffb9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80)` }}>
                            </img>
                            <span>Orlando (FL)</span>
                        </Link>
                        <Link className="city-destination row "
                        to={`city?type=city&city_code=DALLAS&state_code=TX&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1563219125-60d10ffe8877?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1152&q=80)` }}>
                            </img>
                            <span>Dallas (TX)</span>
                        </Link>
                        <Link className="city-destination row "
                        to={`city?type=city&city_code=PHOENIX&state_code=AZ&checkInDate=${checkInDate}&checkOutDate=${checkOutDate}&adult=2&children=0`}>
                            <img className="city-destination-image"
                                style={{ backgroundImage: `url(https://images.unsplash.com/photo-1589046207215-b5ee3097bafc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80)` }}>
                            </img>
                            <span>Phoenix (AZ)</span>
                        </Link>
                    </div>
                </div>
            </div>
            <div className="home-grid">
                <h3 className="home-grid-item-1" >

                    Travel Makes You Happier

                </h3>
                <div className="home-grid-item-2" style={{ backgroundImage: `url(https://images.unsplash.com/photo-1571003123894-1f0594d2b5d9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=449&q=80)` }}>

                </div>
                <h3 className="home-grid-item-3">
                    “I travel because it makes me realize how much I haven’t seen, how much I’m not going to see, and how much I still need to see.” – Carew Papritz
                </h3>
                <div className="home-grid-item-4" style={{ backgroundImage: `url(https://images.unsplash.com/photo-1563911302283-d2bc129e7570?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80)` }}>

                </div>
                <h3 className="home-grid-item-5">
                    Don't Listen to what they say. Go see.
                </h3>
                <div className="home-grid-item-6" style={{ backgroundImage: `url(https://images.unsplash.com/photo-1592555059503-0a774cb8d477?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80)` }}>

                </div>

            </div>

        </div>
    )
}

export default TravelInfo;
