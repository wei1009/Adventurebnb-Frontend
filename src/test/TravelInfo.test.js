import React from "react";
import { render } from "@testing-library/react";
import TravelInfo from "../Home/TravelInfo";
import { MemoryRouter } from "react-router";

// TODO: woefully under-tested!

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <TravelInfo />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});