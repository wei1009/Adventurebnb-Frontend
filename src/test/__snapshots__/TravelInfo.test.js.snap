// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-Info-detail"
  >
    <div
      class="city-destination-recommand"
    >
      <div
        class="container"
      >
        <h2
          class="destination-title"
        >
          Top destinations in the United States
        </h2>
        <div
          class="city-layout column"
        >
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=NEW+YORK&state_code=NY&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1499092346589-b9b6be3e94b2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80);"
            />
            <span>
              New York (NY)
            </span>
          </a>
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=LAS+VEGAS&state_code=NV&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1470076892663-af684e5e15af?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1016&q=80);"
            />
            <span>
              Las Vegas (NV)
            </span>
          </a>
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=LOS+ANGELES&state_code=CA&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1572975165675-ab322e4b55ba?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1171&q=80);"
            />
            <span>
              Los Angeles (CA)
            </span>
          </a>
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=ORLANDO&state_code=FL&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1575089776834-8be34696ffb9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1074&q=80);"
            />
            <span>
              Orlando (FL)
            </span>
          </a>
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=DALLAS&state_code=TX&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1563219125-60d10ffe8877?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1152&q=80);"
            />
            <span>
              Dallas (TX)
            </span>
          </a>
          <a
            class="city-destination row "
            href="/hotelsearch?type=city&city_code=PHOENIX&state_code=AZ&checkInDate=2022-04-01&checkOutDate=2022-04-02&adult=2&children=0"
          >
            <img
              class="city-destination-image"
              style="background-image: url(https://images.unsplash.com/photo-1589046207215-b5ee3097bafc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80);"
            />
            <span>
              Phoenix (AZ)
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="home-grid"
    >
      <h3
        class="home-grid-item-1"
      >
        Travel Makes You Happier
      </h3>
      <div
        class="home-grid-item-2"
        style="background-image: url(https://images.unsplash.com/photo-1571003123894-1f0594d2b5d9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=449&q=80);"
      />
      <h3
        class="home-grid-item-3"
      >
        “I travel because it makes me realize how much I haven’t seen, how much I’m not going to see, and how much I still need to see.” – Carew Papritz
      </h3>
      <div
        class="home-grid-item-4"
        style="background-image: url(https://images.unsplash.com/photo-1563911302283-d2bc129e7570?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80);"
      />
      <h3
        class="home-grid-item-5"
      >
        Don't Listen to what they say. Go see.
      </h3>
      <div
        class="home-grid-item-6"
        style="background-image: url(https://images.unsplash.com/photo-1592555059503-0a774cb8d477?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80);"
      />
    </div>
  </div>
</DocumentFragment>
`;
